api.planx.uk/s3/s3.test.ts

```diff
@@ -0,0 +1,196 @@
+import supertest from "supertest";
+
+import app from "../server";
+
+const mockPutObject = jest.fn(() => ({
+  promise: () => Promise.resolve()
+}))
+
+let getObjectResponse = {};
+
+const mockGetObject = jest.fn(() => ({
+  promise: () => Promise.resolve(getObjectResponse)
+}))
+
+const mockGetSignedUrl = jest.fn(() => {
+  const randomFolderName = "nanoid";
+  const modifiedKey = "modified%20key";
+  return `
+    https://test-bucket.s3.eu-west-2.amazonaws.com/${randomFolderName}/${modifiedKey}?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-SignedHeaders=host
+  `
+});
+
+const s3Mock = () => {
+  return {
+    putObject: mockPutObject,
+    getObject: mockGetObject,
+    getSignedUrl: mockGetSignedUrl,
+  };
+};
+
+jest.mock("aws-sdk/clients/s3", () => {
+  return jest.fn().mockImplementation(() => {
+    return s3Mock();
+  })
+});
+
+jest.mock("nanoid", () => ({
+  ...(jest.requireActual("nanoid")),
+  customAlphabet: jest.fn(() => () => "nanoid")
```

Comment:
Sorry, why do we need this?

Reply:
Good catch - this is leftover from a refactor 👌 Tidied up!